src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -78,31 +139,40 @@
       name: 'approver',
       section: 'requestDetails',
       documentation: `The user that is requested for approval. When set, "group" property is ignored.`,
-      tableCellFormatter: function(approver) {
-        let self = this;
-        this.__subSubContext__.userDAO.find(approver).then((user)=> {
-          if ( user ) {
-            if ( self.__subSubContext__.user.id == user.id ) {
-              self.add(user.legalName);
-            } else {
-              self.add(user.group);
-            }
-          } else {
-            self.add(approver);
-          }
-        });
+      view: function(_, X) {
+        if ( X.data.status === foam.nanos.approval.ApprovalStatus.REQUESTED ) {
+          return {
+            class: 'foam.u2.view.ValueView',
+            data$: X.data$.map((data) => data.REQUESTED)
+          };
+        } else {
+          return { class: 'foam.u2.view.ReferencePropertyView' };
+        }
       },
-      visibility: function(approver) {
-        return approver ?
-          foam.u2.DisplayMode.RO :
-          foam.u2.DisplayMode.HIDDEN;
+      tableCellFormatter: function(approver, data) {
+        let self = this;
+        // If request is REQUESTED, show as Pending
+        // Otherwise, show approver's name
+        if ( data.status === foam.nanos.approval.ApprovalStatus.REQUESTED ) {
+          this.add(data.REQUESTED);
+        } else {
+          this.__subSubContext__.userDAO.find(approver).then(user => {
+            self.add(user ? user.toSummary() : `User #${approver}`);
+          });
+        }
       },
+      visibility: function(status, approver) {
+        if ( status === foam.nanos.approval.ApprovalStatus.REQUESTED || ! approver ) {
+          return foam.u2.DisplayMode.HIDDEN;
```

Comment:
This suggests REQUESTED are HIDDEN?

Reply:
Again was liquid specific code that was added feel free to remove